### PR TITLE
ENH: add http status code expert

### PR DESCRIPTION
**Pull request by schacht-certat**
This adds the module
intelmq.bots.experts.http_status.expert.HttpStatusExpertBot
It also adds a description in the documentation and a changelog entry
introducing the new bot.

### CODE REVIEW

File: intelmq/bots/experts/http_status/expert.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    Parameters
+    ----------
+    field: str
+        The name of the field containing the URL to be checked (required).
+    success_status_codes: List
+        A list of success status codes. If this parameter is omitted or the list is empty,
+        successful status codes are the ones between 200 and 400.
+    """
+    field: str = None  # The field containing the URI
```
Comment (waldbauer-certat) [inline intelmq/bots/experts/http_status/expert.py:25]: Isn't it better if this field would be `url` instead of `field` as its a httpStatusExpert?

**Review by waldbauer-certat — CHANGES_REQUESTED** [review]

File: intelmq/bots/experts/http_status/expert.py (hunk 1 above)
Comment (waldbauer-certat) [reply intelmq/bots/experts/http_status/expert.py:25]: Oh, now I got it. Was a bit confused. Thanks for clarification.

**Review by waldbauer-certat — APPROVED** [review]

File: intelmq/bots/experts/http_status/expert.py (hunk 1 above)
Comment (wagner-certat) [inline intelmq/bots/experts/http_status/expert.py:25]:
```suggestion
    field: str = "source.url"  # The field containing the URI
```
`source.url` would be a good example and meaningful default value

File: intelmq/bots/experts/http_status/expert.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+            event.add('status', 'offline')
+
+            try:
+                response = session.get(event[self.field])
+                # If success_status_codes are configured, we use those
+                # to check the success of the request, otherwise we
+                # rely on the boolean value of the response
+                if (self.success_status_codes and response.status_code in self.success_status_codes) or (response):
+                    event.add('status', "online", overwrite=True)
```
Comment (wagner-certat) [inline intelmq/bots/experts/http_status/expert.py:47]:
```suggestion
                    event.add('status', "online", overwrite=self.overwrite)
```
And add an overwrite parameter

**Review by wagner-certat — CHANGES_REQUESTED** [review]
An `__init__.py` is missing, otherwise `list_all_bots` won't detect it.

Suggestion: As we will have multiple http-related experts, we can group them under `intelmq/bots/experts/http`, so this one could be names `intelmq/bots/experts/http/expert_status.py`.

File: intelmq/bots/experts/http_status/expert.py (hunk 1 above)
Comment (schacht-certat) [reply intelmq/bots/experts/http_status/expert.py]: good idea, fixed!

File: intelmq/bots/experts/http_status/expert.py (hunk 2 above)
Comment (schacht-certat) [reply intelmq/bots/experts/http_status/expert.py]: done.

**Comment by schacht-certat** [comment]
> An `__init__.py` is missing, otherwise `list_all_bots` won't detect it.

also fixed.

> Suggestion: As we will have multiple http-related experts, we can group them under `intelmq/bots/experts/http`, so this one could be names `intelmq/bots/experts/http/expert_status.py`.

still not sure about the best approach, but lets do it like you suggested for now ;)

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**